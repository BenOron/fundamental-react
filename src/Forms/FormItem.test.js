import FormItem from './FormItem';
import { mount } from 'enzyme';
import React from 'react';

describe('FormItem Prop spreading', () => {
    test('should allow props to be spread to the FormItem component', () => {
        const element = mount(<FormItem data-sample='Sample' />);

        expect(
            element.getDOMNode().attributes['data-sample'].value
        ).toBe('Sample');
    });
});
