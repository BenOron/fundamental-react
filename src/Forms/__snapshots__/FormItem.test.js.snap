// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormItem /> create form item 1`] = `
<div
  className="fd-form-item"
>
  <label
    aria-required={true}
    className="fd-form-label fd-form-label--required"
    forAttr="input-1"
  >
    Default Input
  </label>
  <input
    className="fd-input"
    id="input-1"
    placeholder="Field placeholder text"
    state="warning"
    type="text"
  />
  <textarea
    className="fd-textarea blue"
    id="textarea-1"
  >
    Pellentesque metus lacus commodo eget justo ut rutrum varius nunc.
  </textarea>
  <textarea
    className="fd-textarea"
    id="textarea-2"
  >
    Pellentesque metus lacus commodo eget justo ut rutrum varius nunc.
  </textarea>
  <select
    className="fd-form-select"
    id="select-1"
  >
    <option
      value="1"
    >
      Duis malesuada odio volutpat elementum
    </option>
    <option
      value="2"
    >
      Suspendisse ante ligula
    </option>
    <option
      value="3"
    >
      Sed bibendum sapien at posuere interdum
    </option>
  </select>
  <select
    className="fd-form-select blue"
    disabled={true}
    id="select-1"
  >
    <option
      value="1"
    >
      Duis malesuada odio volutpat elementum
    </option>
  </select>
</div>
`;
